nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-api/src/main/java/org/apache/nifi/web/api/dto/EntityFactory.java

```diff
@@ -77,6 +77,8 @@ public ProcessorEntity createProcessorEntity(final ProcessorDTO dto, final Revis
         if (dto != null) {
             entity.setPermissions(permissions);
             entity.setStatus(status);
+            status.setCanRead(permissions.getCanRead());
```

Comment:
I believe the potential for a NPE was introduce here. Permissions doesn't get checked for null until line 85.

This also applies to the other changes in this file.